Update pick-and-place uArm workshop (phases 1-6) (#5277)
viamrobotics/docs: docs/tutorials/pick-and-place/05-perception-guided-picking.md

{{< expand "Reveal the grasp descent" >}}
Think of both numbers as a height above the block's center:

- Standoff: 100 mm above center (`APPROACH_MM`)
- Grasp: TCP needs to stop 60 mm above center (`GRIPPER_LENGTH_MM`)

The gripper still has 100 − 60 = 40 mm to descend:

```python
grasp_distance = (APPROACH_MM - GRIPPER_LENGTH_MM) * -1  # 40 mm
```

Both constants are negative because they're measured upward from the block, so APPROACH_MM - GRIPPER_LENGTH_MM comes out to −40 mm; the \* -1 flips it into a positive 40 mm downward move.

Sanity check: using APPROACH_MM alone would drive the full 100 mm down and crash through the block; using zero would never leave the standoff. 40 mm is exactly the gap between the two.

{{< /expand >}}

{{< checkpoint >}}
Before wiring up the moves, print `approach_pose` and compare its `z` to `obj_in_cam.pose.z`: the approach `z` should sit about 100 mm toward the camera.
{{< /checkpoint >}}

## Descend in the gripper frame

> The approach move runs first, tagged `reference_frame="cam-1"`. At that instant the arm is still at `home-pose`, so `cam-1` sits exactly where the detection assumed and the motion service resolves the standoff correctly. The trouble only starts with the _second_ move: once the arm travels to the standoff it carries `cam-1` with it, so any further pose you tag `cam-1` resolves against the camera's new position, not the one your offset math used.

Reviewer: ```suggestion
The approach move runs first, tagged `reference_frame="cam-1"`. At that instant the arm is still at `home-pose`, so `cam-1` sits exactly where the detection assumed and the motion service resolves the standoff correctly. This assumption shifts after that move; once the arm travels to the standoff it carries `cam-1` with it, so any further pose you tag `cam-1` resolves against the camera's new position, not the one your offset math used.
```
Slight tweak in prose. Indicating "trouble" before any issue has occurred is distracting to the point of this section.

> Rather than chase the camera frame, re-anchoring the block pose after every move, you sidestep it. The gripper is now parked directly above the block at the standoff, already lined up to come straight down. All that is left is to drop the last stretch, and that distance is a property of the gripper, not the camera: descend `grasp_distance` millimeters along the gripper's own `z` axis.

Reviewer: ```suggestion
Rather than chase the camera frame, re-anchoring the block pose after every move, you sidestep it. The gripper is now parked directly above the block at the standoff, already lined up to come straight down. All that is left is to drop the last stretch, and we can describe that move in the frame of the gripper instead of the camera: descend `grasp_distance` millimeters along the gripper's own `z` axis.
```
